dask/array/tests/test_array_core.py

```diff
@@ -1204,6 +1204,14 @@ def test_reshape_unknown_dimensions():
     pytest.raises(ValueError, lambda: da.reshape(a, (-1, -1)))
 
 
+def test_reshape_avoids_large_chunks():
+    limit = parse_bytes("128MiB")
+    x = da.random.random((300, 180, 4, 18483), chunks=(-1, -1, 1, 183))
+    result = x.reshape(300, 180, -1)
+    nbytes = x.dtype.itemsize
+    assert (np.prod(result.chunksize) * nbytes) < (limit)
```

Comment:
I think we need an example where the chunks tuple along a couple of axes are equal to test decrementing the Counter.
Something like this assuming that these chunks are also too big.
```python
da.random.random((300, 300 4, 18483), chunks=(30, 30, 1, 1830))
```

Reply:
I've added a second test (with pytest parametrize) to do this.